src/installer/utils.py

```diff
@@ -101,7 +109,9 @@ def parse_wheel_filename(filename: str) -> WheelFilename:
     wheel_info = _WHEEL_FILENAME_REGEX.match(filename)
     if not wheel_info:
         raise ValueError(f"Not a valid wheel filename: {filename}")
-    return WheelFilename(*wheel_info.groups())
+    parsed = wheel_info.groups()
+    normalized_name = normalize_distribution_name(parsed[0])
```

Comment:
Sadly, we don't know for sure if the name inside the wheel will be canonicalized.

For example, `zope.interface`'s wheels contain `zope.interface-5.4.0.dist-info` (and not `zope-interface`) thanks, in no small part, to setuptools' behaviour in this area.

Reply:
Oh, wait, that's what this is doing. :)

Reply:
So, we're going to either need to probe a few names, or search the directory listing.

Reply:
I see pip searches the zip for a `.dist-info`. So I guess we should do that.

Reply:
I agree. Let's close this out, since we've established that merely canonicalising the name isn't sufficient.